src/requests/compat.py

```diff
@@ -10,6 +10,14 @@
 import importlib
 import sys
 
+# -------
+# urllib3
+# -------
+from urllib3 import __version__ as urllib3_version
+
+# Detect which major version of urllib3 is being used.
+is_urllib3_2 = urllib3_version.split('.')[0] == 2
```

Comment:
I believe this is a string, not an integer.

Reply:
Good catch on the `int`. For the equivalence, I don't know if `>=` is what we want if this is specifically scoping the 2.x major version. In the same way I would want an `is_py3` to include Python 4.x. We could something like `is_gt_urllib3_1` but that seems like it may be premature forwards-compatibility?

Right now we're scoped at `urllib3<3` and if we add this check to any other behaviors, I'd rather they be scoped to the major version. That will let tests fail if we major version again and we can make an informed decision at that point when adding support. Otherwise, we may unintentionally carry forward behaviors that are subtly wrong.

I can see similar risks with both sides, so not a hill I'm going to die on, but that was my initial thought process.

Reply:
I've changed the check from checking for urllib3 2.x to check for 1.x. That leaves us open to forward compatibility without the confusing behavior with `is_urllib3_2` including newer major versions.